platform/linux/src/headless_backend_egl.cpp

```diff
@@ -0,0 +1,75 @@
+#include <mbgl/platform/default/headless_backend.hpp>
+#include <mbgl/platform/default/headless_display.hpp>
+
+#include <mbgl/platform/log.hpp>
+
+#include <EGL/egl.h>
+
+#include <cassert>
+
+namespace mbgl {
+
+struct EGLImpl : public HeadlessBackend::Impl {
+    EGLImpl(EGLContext glContext_, EGLDisplay display_)
+            : glContext(glContext_),
+              display(display_) {
+    }
+
+    ~EGLImpl() {
+        if (glContext != eglGetCurrentContext()) {
+            activateContext();
+        }
+        if (!eglDestroyContext(display, glContext)) {
+            throw std::runtime_error("Failed to destroy EGL context.\n");
+        }
+    }
+
+    void activateContext() final {
+        if (!eglMakeCurrent(display, EGL_NO_SURFACE, EGL_NO_SURFACE, glContext)) {
+            throw std::runtime_error("Switching OpenGL context failed.\n");
+        }
+    }
+
+    void deactivateContext() final {
+        if (!eglMakeCurrent(display, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT)) {
+            throw std::runtime_error("Removing OpenGL context failed.\n");
+        }
+    }
+
+    EGLContext glContext = EGL_NO_CONTEXT;
+    EGLDisplay display = EGL_NO_DISPLAY;
+};
+
+gl::glProc HeadlessBackend::initializeExtension(const char* name) {
+    return eglGetProcAddress(name);
+}
+
+bool HeadlessBackend::hasDisplay() {
+    if (!display) {
+        display.reset(new HeadlessDisplay);
+    }
+    return bool(display);
+};
+
+void HeadlessBackend::createContext() {
+    assert(!hasContext());
+    assert(hasDisplay());
+
+    EGLDisplay display_ = display->attribute<EGLDisplay>();
+    EGLConfig& config = display->attribute<EGLConfig&>();
+
+    if (!eglBindAPI(EGL_OPENGL_API)) {
```

Comment:
You are setting `MBGL_USE_GLES2` above, but requesting a desktop OpenGL API here. Did you try requesting `EGL_OPENGL_ES_API`?